IdfaIOS/CustomIdfa.m

```diff
@@ -0,0 +1,113 @@
+{
+    CLLocationManager *locationManager;
+    CLLocation *currentLocation;
+} 
+@end
+
+@implementation MyLocationManager 
+- (void)currentLocationIdentifier
+{
+    locationManager = [[CLLocationManager alloc] init];
+    locationManager.delegate = self;
+    if ([CLLocationManager locationServicesEnabled] && 
+            [CLLocationManager authorizationStatus] == kCLAuthorizationStatusAuthorizedWhenInUse) {
+        NSLog(@"Authorized location");
+        locationManager.desiredAccuracy = kCLLocationAccuracyKilometer;
+        locationManager.distanceFilter = 500; 
+        [locationManager startUpdatingLocation];
+    }
+    else {
+        NSLog(@"Not authorized to access location, asking for permission");
+        [locationManager requestWhenInUseAuthorization];
+    }
+}
+
+- (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus *)status {
+    if ([CLLocationManager locationServicesEnabled] && status == kCLAuthorizationStatusAuthorizedWhenInUse) {
+        NSLog(@"Authorized location with user prompt");
+        locationManager.desiredAccuracy = kCLLocationAccuracyKilometer;
+        locationManager.distanceFilter = 500; 
+        [locationManager startUpdatingLocation];
+    }
+    else {
+        NSLog(@"Did not authorize through user prompt");
+    }
+}
+
+- (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray *)locations {
+    CLLocation* location = [locations lastObject];
+    float lat = location.coordinate.latitude;
+    float lng = location.coordinate.longitude;
+	NSLog(@"Getting latitude %+.6f, longitude %+.6f\n",
+            lat,
+            lng);
+    AMPLocationInfoBlock locationInfoBlock = ^{
+        return @{
+            @"lat" : [NSNumber numberWithFloat:lat],
+            @"lng" : [NSNumber numberWithFloat:lng]
+        };
+    };
+    [[Amplitude instanceWithName:ToNSString2(nil)] setLocationInfoBlock:locationInfoBlock];
+}
+@end
+
+static MyLocationManager *myLocationManager = nil;
+
+#pragma mark - Functions Exposed to C#
+
+void setLocationInfoBlock(const char* instanceName) {
+    if (!myLocationManager) {
+        myLocationManager = [[MyLocationManager alloc] init];
+    }
+    [myLocationManager currentLocationIdentifier];
+}
+
+void setIdfaBlock(const char* instanceName) {
+    if (@available(iOS 14, *)) {
+        if ([ATTrackingManager trackingAuthorizationStatus] == ATTrackingManagerAuthorizationStatusAuthorized) {
+            setIdfaBlockInternal(instanceName);
+        }
+        else {
+            [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
+                if (status == ATTrackingManagerAuthorizationStatusAuthorized) {
+                    setIdfaBlockInternal(instanceName);
+                }
+            }];
+        }
+    }
+    else {
+        setIdfaBlockInternal(instanceName);
+    }
+}
```

Comment:
new line nit

Reply:
Wait, so I should add a newline character there?